### DISCUSSION TITLE
Background Saves???

### DISCUSSION THREAD

**Discussion by K10ChevyGuy** [kg4vdk/arcos-linux-modules#20] [It's Not In The Manual...Promise!] [answered]
I seem to have the persistent-storage operating properly. I have multiple settings being saved Firefox logs me in to settings and preferences I set. Seems to work good for most things. First, I cannot seem to persistently save a different background than the default blue arcOS, I would prefer to use the "olive" arcOS background. I have to manually set that each time I boot the OS. Second, On more than one computer the video brightness seems to be initially set to approx 35 or 40 and has to be adjusted up for my preferences. Again it's does not seem to be a persistent storage item. I must set it each time I do a start up. 

I guess the questions are; 
     How do I get the background to remain through to next boot?
     How do I set the brightness to be persistent?

I will keep working from this end to see if I can find a fix or workaround but so far nothing solid.
Great job on doing what you have done to this point thank you for putting this together for the Amateur Radio Community!

**Comment by kg4vdk** [comment]
For the background image, place any .jpg you like inside your CALLSIGN directory, and make sure it is named "wallpaper.jpg".

If you like the olive version, you can copy it from /opt/arcOS/backgrounds and rename it. When "Station Setup" runs, it will set wallpaper.jpg as your background. 

This is something I mentioned in a video, just yesterday...at the 4:20 mark. Maybe that makes it the "high" point of the [video](https://youtu.be/KToPZhVzphw?feature=shared). :-)

For the brightness, this is a perfect application of a USER module/script. 

I'm not certain of the commands needed to set the brightness off the top of my head (and I'm not at a machine right now). I'll do some looking and get back to you on that part. If you come across something that works from the command line, just throw it in a Bash script, and drop it into the USER module directory. Any script there that ends in `.sh` will be run (after CORE and COMMUNITY) when you press "OK" during Station Setup.

**Reply by K10ChevyGuy** [reply]
Background fix worked great! Thank you for that!

**Comment by kg4vdk** [comment]
Here's what worked for me on a ThinkPad X1 Carbon (6th Gen)...

Saved as `/media/user/ARCOS-DATA/QRV/KG4VDK/arcos-linux-modules/USER/00_BRIGHTNESS.sh`:

```
#!/bin/bash

echo "200" | sudo tee /sys/class/backlight/intel_backlight/brightness
```

If that file doesn't exist on your machine, take a look around in `/sys/class/backlight` for something similarly relevant to your machine.

I was able to use `cat /sys/class/backlight/intel_backlight/max_brightness` to find the maximum allowable setting. In my case this was 1060. Change the "200" to suit your needs. I also learned not to set it to 0...I'm pretty good with a non-GUI environment, but an all black screen was a challenge. :-)

**Reply by K10ChevyGuy** [reply]
I don't have intel chip set in both computers that I use so I am working up a fix using acpi_video0 as the option to address the video brightness. It's working partially. I will share more when I get it better tuned.

**Reply by kg4vdk** [reply]
Great to hear! If I can be of any assistance, let me know.

**Comment by K10ChevyGuy** [comment, marked as answer]
OK. So... I lack the thought process at the moment to run this without calling it from another script to sudo it. Maybe you can set this up to be a self contained script ran with sudo permissions.
The script is designed to look for the intel_backlight directory, if available it sets brightness to approx 80% (that's the '1250' (out of a max of 1500). If it does not find the intel_backlight directory it confirms acpi_video0 is present  then if present it sets the brightness to again approx 90% that's the '18' (out of a max of 20)
It works on my Predator laptop (intel_backlight) and my Panasonic Toughbook (acpi_video0).

All that being said. there is most likely a more elegant solution. but this one seems to work well for me.

I pulled another script from an ubuntu forum and hacked it into this. It is very likely this code could use some clean up.
I cannot recall the author of the original script but If I find it I will add credit for that in a future post.

> ------------------------------------------------------------
> set_brightness.sh
> ------------------------------------------------------------
> #!/bin/bash
> 
> FILE=/sys/class/backlight/intel_backlight
> if [ -d "$FILE" ]; then
>     TARGET="intel_backlight" && LOGIC='1250'
> else 
>     TARGET="acpi_video0" && LOGIC='18'
> fi
> > cd /sys/class/backlight
> MAX="$(cat "${TARGET}/max_brightness")"
> 
> for i in */; do
> 	if [[ "${TARGET}/" != "$i" && -e "${i}brightness" ]]; then
> 		cat "${i}max_brightness" > "${i}brightness"
> 	fi
> done
> echo "$LOGIC" > "${TARGET}/brightness"

**Reply by kg4vdk** [reply]
Great work! To deal with needing sudo privileges to `echo` or `cat` into root owned files...look into using something like:

`echo "foo" | sudo tee /path/to/file`

This works like `echo "foo" > /path/to/file` but with root privileges. 

If you ever need to append something to a root owned file, like `echo "foo" >> ...`, the `tee` command has the `--append` option available.
